src/appforms/src/core/040-Model06Submission.js

```diff
@@ -379,6 +386,34 @@ appForm.models = function(module) {
 
     this.setFormFields(formFields);
   };
+
+  /**
+   * Submission.prototype.pruneRemovedFields - Pruning fields that have been deleted or removed from the form
+   *
+   * @return {type}  description
+   */
+  Submission.prototype.pruneRemovedFields = function(cb) {
+    var that = this;
+    var formFields = this.getFormFields();
+    var newFields = [];
+
+    this.getForm(function(err, form) {
+      if (err) {
+        return cb(err);
+      }
+      //Loop and push matching fields
+      _.each(formFields, function(field) {
+        var fieldId = field.fieldId;
+        if (form.fields.hasOwnProperty(fieldId)) {
+          newFields.push(field);
```

Comment:
@nialldonnellyfh - Ok I've updated it now to remove related files also. I didn't think about left over files. 👍